src/main/java/org/radarcns/webapp/filter/AuthorizationFeature.java

```diff
@@ -0,0 +1,31 @@
+package org.radarcns.webapp.filter;
+
+import java.lang.reflect.Method;
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.ext.Provider;
+import org.radarcns.auth.NeedsPermission;
+import org.radarcns.auth.NeedsPermissionOnProject;
+import org.radarcns.auth.NeedsPermissionOnSubject;
+import org.radarcns.auth.PermissionFilter;
+import org.radarcns.auth.PermissionOnProjectFilter;
+import org.radarcns.auth.PermissionOnSubjectFilter;
+
+/** Authorization for different auth tags. */
+@Provider
+public class AuthorizationFeature implements DynamicFeature {
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext context) {
+        Method method = resourceInfo.getResourceMethod();
+        if (method.isAnnotationPresent(NeedsPermission.class)) {
+            context.register(PermissionFilter.class, 2000);
+        }
+        if (method.isAnnotationPresent(NeedsPermissionOnProject.class)) {
+            context.register(PermissionOnProjectFilter.class, 2050);
+        }
+        if (method.isAnnotationPresent(NeedsPermissionOnSubject.class)) {
+            context.register(PermissionOnSubjectFilter.class, 2100);
```

Comment:
How was the priority decided here?

Reply:
In the end, it is just an in-app ordering
1. Cors (500)
2. Authentication (1000)
3. Authorization (permission only) (2000)
4. Authorization (permission + project) (2050)
5. Authorization (permission + project) (2100)

It ensures the order, and leaves plenty of room for other filters to be added in between.

Reply:
Okay great. Typo there (permission + subject) but got it! Thanks